Comments/ClientApp/src/components/LoginPanel/LoginPanel.js

```diff
@@ -0,0 +1,222 @@
+	questionsTabIsOpen: boolean,
+}
+
+type StateType = {
+	respondingAsOrg: boolean | null,
+	respondingAsOrgType: string | null,
+}
+
+export class LoginPanel extends Component<PropsType, StateType> {
+
+	constructor(props: PropsType) {
+		super(props);
+
+		this.state = {
+			respondingAsOrg: null,
+			respondingAsOrgType: null,
+		};
+	}
+
+	componentDidMount() {}
+
+	fieldsChangeHandler = (e) => {
+		let fieldValue = e.target.value;
+
+		if (fieldValue === "true" || fieldValue === "false") {
+			fieldValue = (fieldValue === "true");
+		}
+
+		this.setState({	[e.target.name]: fieldValue });
+	};
+
+	goBack = () => {
+		const { respondingAsOrgType } = this.state;
+
+		// back to second question
+		let newState = { respondingAsOrgType: null };
+
+		// back to first question
+		if (respondingAsOrgType === null || respondingAsOrgType === "code") {
+			newState.respondingAsOrg = null;
+		}
+
+		this.setState(newState);
+	};
+
+	render() {
+		const { respondingAsOrg, respondingAsOrgType } = this.state;
+		const { organisationalCommentingFeature, questionsTabIsOpen } = this.props;
+
+		const showFirstScreen = (respondingAsOrg === null);
+		const showSecondScreen = (respondingAsOrg === true && (respondingAsOrgType === null || respondingAsOrgType === "code"));
+		const showLogin = (respondingAsOrg === false || respondingAsOrgType !== null || !organisationalCommentingFeature);
+		const showBackLink = (!showFirstScreen);
+
+		const loginBannerProps = {
+			signInButton: false,
+			signInText: "to access the consultation",
+			currentURL: this.props.match.url,
+			signInURL: this.context.signInURL,
+			registerURL: this.context.registerURL,
+			allowOrganisationCodeLogin: false,
+			title: "Make or review comments as an individual",
+			isInCommentsPanel: true,
+		};
+
+		if (respondingAsOrgType === "lead" || respondingAsOrgType === "sole") {
+			loginBannerProps.title = "Make or review comments on behalf of your organisation";
+		}
+
+		if (respondingAsOrgType === "code") {
+			loginBannerProps.allowOrganisationCodeLogin = true;
+			loginBannerProps.codeLoginOnly = true;
+			loginBannerProps.title = null;
+		}
+
+		if (!organisationalCommentingFeature) {
+			loginBannerProps.title = null;
+			loginBannerProps.signInText = null;
+		}
+
+		return (
+			<>
+				{organisationalCommentingFeature &&
+					<>
+						{showFirstScreen &&
+							<LoginSelectionOrg
+								currentlySelected={respondingAsOrg}
+								onChangeOption={this.fieldsChangeHandler}
+							/>
+						}
+
+						{showSecondScreen &&
+							<LoginSelectionCode
+								currentlySelected={respondingAsOrgType}
+								onChangeOption={this.fieldsChangeHandler}
+							/>
+						}
+					</>
+				}
+
+				{showLogin &&
+					<LoginBannerWithRouter {...loginBannerProps} />
+				}
+
+				{showBackLink &&
+					<div className="container">
+						<span role="button" className="login-panel__back" id="loginPanelBackButton" tabIndex={0} onMouseDown={this.goBack}>&lt; Back</span>
```

Comment:
i like that you've used role=button.

just a question though - why not use an anchor tag? is it just for styling purposes?

feel free to keep the span though.

Reply:
Eslint was kicking off about it being an anchor without a href. It suggested using a button; but I wasn't keen on unstyling it to look like a link, so used the span. It did feel a bit hacky, but probably the best way forward without having to change the implementation...?

https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-is-valid.md

Reply:
how about href="#"?

i see that's one of the options in that link that it tells you not to do though. which is fair, a button element would be better, but i know what you mean about unstyling it.  do you think the design system should support styling a button as a link?

Reply:
btw, can you use onClick instead of onMouseDown please? just for consistency with the other buttons.

Reply:
Yeah, it had an issue with onClick being used on spans:

Visible, non-interactive elements with click handlers must have at least one keyboard listener.
https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/click-events-have-key-events.md

I googled it and someone suggested using onMouseDown instead, which seemed to fix it.

I think changing it to a button would probably be best and I'll just override the styling to make it look like a link. I did think that there was a variant for buttons to look like links, but there wasn't. I know BS had/has `btn-link`, so maybe there's a case for it...